fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/TransactionImpl.java

```diff
@@ -76,20 +112,31 @@ public boolean isShortLived() {
 
     @Override
     public void expire() {
-        // TODO Auto-generated method stub
+        this.expiration = Instant.now();
+        this.expired = true;
+    }
 
+    @Override
+    public boolean hasExpired() {
+        if (this.expired) {
+            return true;
+        }
+        this.expired = this.expiration.isBefore(Instant.now());
+        return this.expired;
     }
 
     @Override
     public Instant updateExpiry(final Duration amountToAdd) {
-        // TODO Auto-generated method stub
-        return null;
+        failIfExpired();
```

Comment:
Should we allow the expiration to be update if the transaction has already expired?

Reply:
no i don't think so. We should probably be rolling back expired transactions on a timer over in the transaction manager rather than exclusively waiting for a request against it